app/workflows/judge_assign_task_creator.rb

```diff
@@ -22,6 +25,15 @@ def call
 
   attr_reader :appeal, :judge
 
+  def reassign_existing_open_task(open_judge_assign_task)
+    open_judge_assign_task.reassign({
+                                      assigned_to_type: @judge.class.name,
+                                      assigned_to_id: @judge.id,
+                                      appeal: appeal
+                                    }, current_user)
+    @task = @appeal.tasks.open.find_by_type(:JudgeAssignTask)
```

Comment:
I think you can reduce the number of queries by removing line 34 and replacing the implementation of the `task` method with `find-or-create` code.

Alternatively, you can do something like this if desired:
```suggestion
    new_task, old_task, new_children = open_judge_assign_task.reassign({
                                      assigned_to_type: @judge.class.name,
                                      assigned_to_id: @judge.id,
                                      appeal: appeal
                                    }, current_user)
    new_task
```

Reply:
I refactored to something like the second option.  Also, while refactoring this class, I noticed that the `JudgeAssignTaskCreator` `task` method does not seem to be used anywhere in our codebase.  Curious.

Reply:
Do you can what the return value of this method is -- it's not used anywhere? If not, remove the last line.